### ISSUE TITLE
Crash in parsing in prometheus metrics 

### ISSUE THREAD

**Issue by johnhtodd** [dmachard/go-dnscollector#509]
**Describe the bug**
Malformed DNS data causes a crash in the Prometheus sections of code.  I am using the pipeline branch, but I suspect that is not relevant.

**To Reproduce**
Difficult to say other what the exact query was that caused the crash - I wasn't recording this data anywhere else, other than what you see on the output in the crash.

**Expected behavior**
Hoping for no crash.  :-)

**Additional context**

My prometheus configuration

```
pipelines:
  - name: prom
    prometheus:
      listen-ip: 0.0.0.0
      listen-port: 8081
      basic-auth-enable: false
      prometheus-prefix: "dnscollector"
      top-n: 100
      histogram-metrics-enabled: true
      prometheus-labels: ["stream_id"]
```

The crash:

```
INFO: 2023/12/14 02:45:03.397202 [dnsdist-from-outside] processor=dnstap#1 - waiting dns message to process...
INFO: 2023/12/14 02:45:03.399139 [dnsdist-from-outside] collector=dnstap#1 - receiver framestream initialized
panic: label value "dr._dns-sd._udp. \xbf\x9f\x01" is not valid UTF-8

goroutine 78 [running]:
github.com/prometheus/client_golang/prometheus.MustNewConstMetric(...)
	/root/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/value.go:129
github.com/dmachard/go-dnscollector/loggers.(*PrometheusCountersSet).Collect(0xc0001343c0, 0xc000475380?)
	/home/jtodd/go-dnscollector/go-dnscollector-pipeline-branch/go-dnscollector/loggers/prometheus.go:528 +0x1aed
github.com/prometheus/client_golang/prometheus.(*wrappingCollector).Collect.func1()
	/root/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/wrap.go:129 +0x28
created by github.com/prometheus/client_golang/prometheus.(*wrappingCollector).Collect in goroutine 77
	/root/go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/wrap.go:128 +0x7d
root@ub20template:/home/jtodd/go-dnscollector/go-dnscollector-pipeline-branch/go-dnscollector#

```

**Comment by dmachard** [comment]
I don't success to reproduce with invalid name :(
Is it possible to provide the complete query name or share a pcap file?

**Comment by johnhtodd** [comment]
Sorry, I don't have the packet in question - I wasn't capturing at that point, and haven't been able to reproduce yet.

**Comment by johnhtodd** [comment]
It just happened again, and I now figured out what the trigger is: when a pull is made against my Prometheus endpoint, the crash occurs. It is not happening when the packet reaches go-dnscollector - it is only when I do "curl http://127.0.0.1:8081/metrics" that the crash happens.

Same general failure, but the line is:
panic: label value "lb._dns-sd._udp.\xd0\xdfP\x01" is not valid UTF-8

**Comment by dmachard** [comment]
I suspected that, but for my part, the crash did not occur when I do the curl. Instead, the metrics are truncated with is not normal too.
Before fixing I would like to reproduce and write a test unit.

Thanks to report that.

**Comment by dmachard** [comment]
Panic error reproduced ! 
This issue is related to the prometheus logger only.

**Comment by dmachard** [comment]
fix pushed in main and pipeline branch

**Comment by dmachard** [comment]
Fixed in v0.39.0.
